src/crimson/osd/recovery_backend.cc

```diff
@@ -297,29 +297,29 @@ RecoveryBackend::handle_scan_get_digest(
     [this, query_epoch=m.query_epoch, conn
     ](auto backfill_interval) {
       auto reply = crimson::make_message<MOSDPGScan>(
-	MOSDPGScan::OP_SCAN_DIGEST,
+	MOSDPGScan::OP_SCAN_DIGEST_REPLY,
 	pg.get_pg_whoami(),
 	pg.get_osdmap_epoch(),
 	query_epoch,
 	spg_t(pg.get_info().pgid.pgid, pg.get_primary().shard),
 	backfill_interval.begin,
-	backfill_interval.end);
+	backfill_interval.end,
+	backfill_interval.version);
       encode(backfill_interval.objects, reply->get_data());
       return conn->send(std::move(reply));
     });
 }
 
 RecoveryBackend::interruptible_future<>
-RecoveryBackend::handle_scan_digest(
+RecoveryBackend::handle_scan_digest_reply(
   MOSDPGScan& m)
 {
   logger().debug("{}", __func__);
   // Check that from is in backfill_targets vector
   ceph_assert(pg.is_backfill_target(m.from));
 
   BackfillInterval bi(m.begin, m.end);
-  //XXX: See following commits to explain the empty version
-  bi.populate(m.get_data(), eversion_t());
+  bi.populate(m.get_data(), m.version);
```

Comment:
Why do we need the version here?  We won't submit any IOs on the objects in the range we're scanning, so I don't think we need to update the replica bi's from the log/projected_log (not true of the primary).

Alternately, if we actually do need the version, why is it safe to pass in m.version here from an old replica which didn't populate it?

Reply:
> why is it safe to pass in m.version here from an old replica which didn't populate it?

My bad there, version was omitted from MOSDPGScan